lib/src/gitignore.rs

```diff
@@ -14,189 +14,99 @@
-            Some(rest) => (true, rest),
-        };
-        is_rooted |= input.contains('/');
-
-        let mut regex = String::new();
-        regex.push('^');
-        regex.push_str(prefix);
-        if !is_rooted {
-            regex.push_str("(.*/)?");
-        }
-
-        let components = input.split('/').collect_vec();
-        for (i, component) in components.iter().enumerate() {
-            if *component == "**" {
-                if i == components.len() - 1 {
-                    regex.push_str(".*");
-                } else {
-                    regex.push_str("(.*/)?");
-                }
-            } else {
-                let mut in_escape = false;
-                let mut character_class: Option<String> = None;
-                for c in component.chars() {
-                    if in_escape {
-                        in_escape = false;
-                        if !matches!(c, ' ' | '#' | '!' | '?' | '\\' | '*') {
-                            regex.push_str(&regex_escape("\\"));
-                        }
-                        regex.push_str(&regex_escape(&c.to_string()));
-                    } else if c == '\\' {
-                        in_escape = true;
-                    } else if let Some(characters) = &mut character_class {
-                        if c == ']' {
-                            regex.push('[');
-                            regex.push_str(characters);
-                            regex.push(']');
-                            character_class = None;
-                        } else {
-                            characters.push(c);
-                        }
-                    } else {
-                        in_escape = false;
-                        if c == '?' {
-                            regex.push_str("[^/]");
-                        } else if c == '*' {
-                            regex.push_str("[^/]*");
-                        } else if c == '[' {
-                            character_class = Some(String::new());
-                        } else {
-                            regex.push_str(&regex_escape(&c.to_string()));
-                        }
-                    }
-                }
-                if in_escape {
-                    regex.push_str(&regex_escape("\\"));
-                }
-                if i < components.len() - 1 {
-                    regex.push('/');
-                }
-            }
-        }
-        if matches_only_directory {
-            regex.push_str("/.*");
-        } else {
-            regex.push_str("(/.*|$)");
-        }
-        let regex = Regex::new(&regex).unwrap();
-
-        Some(GitIgnoreLine { is_negative, regex })
-    }
-
-    fn matches(&self, path: &str) -> bool {
-        self.regex.is_match(path)
-    }
+use std::{fs, io};
+
+use ignore::gitignore;
+use thiserror::Error;
+
+#[derive(Debug, Error)]
+pub enum GitIgnoreError {
+    #[error("failed to read ignore patterns from file {path}: {source}")]
```

Comment:
nit: capitalize "Failed", and remove `: {source}` (which will be printed by CLI.)